### PR TITLE
Samba 4.16.5

### PR DESCRIPTION
**Pull request by DavidePrincipi**
Refs https://github.com/NethServer/dev/issues/6702

### CODE REVIEW

**Comment by DavidePrincipi** [comment]
Samba services start, but if I run

    ldapsearch -ZZ -H ldap://<IP of DC> -D '' -w '' -b '' -s base
    
then I get in `journalctl -M nsdc`:

    Sep 28 23:13:11 nsdc-nscom1.ad.dp.nethserver.net samba[27]: _tstream_tls_accept_send: TLS ../../source4/lib/tls/tls_tstream.c:1300 - The request is invalid.. Failed to set default priorities

**Comment by DavidePrincipi** [comment]
We should add a gnutls.config file to nethserver-dc RPM

See https://lists.samba.org/archive/samba/2020-December/233651.html


Thanks to @gsanchietti

**Comment by DavidePrincipi** [comment]
TLS is working now. This is a diff of available ciphers

```diff
@@ -3,67 +3,46 @@
 | ssl-enum-ciphers: 
 |   TLSv1.0: 
 |     ciphers: 
-|       TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_3DES_EDE_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_RC4_128_MD5 - strong
-|       TLS_RSA_WITH_RC4_128_SHA - strong
 |     compressors: 
 |       NULL
 |   TLSv1.1: 
 |     ciphers: 
-|       TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_3DES_EDE_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_128_CBC_SHA - strong
 |       TLS_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_RC4_128_MD5 - strong
-|       TLS_RSA_WITH_RC4_128_SHA - strong
 |     compressors: 
 |       NULL
 |   TLSv1.2: 
 |     ciphers: 
-|       TLS_DHE_RSA_WITH_3DES_EDE_CBC_SHA - strong
 |       TLS_DHE_RSA_WITH_AES_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_AES_128_CBC_SHA256 - strong
+|       TLS_DHE_RSA_WITH_AES_128_CCM - strong
 |       TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 - strong
 |       TLS_DHE_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 - strong
+|       TLS_DHE_RSA_WITH_AES_256_CCM - strong
 |       TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA256 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_128_GCM_SHA256 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA256 - strong
-|       TLS_DHE_RSA_WITH_CAMELLIA_256_GCM_SHA384 - strong
-|       TLS_RSA_WITH_3DES_EDE_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA - strong
+|       TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 - strong
 |       TLS_RSA_WITH_AES_128_CBC_SHA - strong
-|       TLS_RSA_WITH_AES_128_CBC_SHA256 - strong
+|       TLS_RSA_WITH_AES_128_CCM - strong
 |       TLS_RSA_WITH_AES_128_GCM_SHA256 - strong
 |       TLS_RSA_WITH_AES_256_CBC_SHA - strong
-|       TLS_RSA_WITH_AES_256_CBC_SHA256 - strong
+|       TLS_RSA_WITH_AES_256_CCM - strong
 |       TLS_RSA_WITH_AES_256_GCM_SHA384 - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_128_CBC_SHA256 - strong
-|       TLS_RSA_WITH_CAMELLIA_128_GCM_SHA256 - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA - strong
-|       TLS_RSA_WITH_CAMELLIA_256_CBC_SHA256 - strong
-|       TLS_RSA_WITH_CAMELLIA_256_GCM_SHA384 - strong
-|       TLS_RSA_WITH_RC4_128_MD5 - strong
-|       TLS_RSA_WITH_RC4_128_SHA - strong
 |     compressors: 
 |       NULL
 |_  least strength: strong
```

**Comment by DavidePrincipi** [comment]
In new domain provisioning procedure:

    Sep 29 11:22:36 nsdc-nscom1.ad.dp.nethserver.net samba-tool[30]: ERROR(ldb): uncaught exception - 0000052D: Constraint violation - check_password_restrictions: the password does not meet the complexity criteria!

File: root/etc/e-smith/events/actions/nethserver-dc-install [hunk 1]
```diff
@@ -37,7 +37,7 @@ if [[ ! -x ${nsroot}/usr/sbin/samba ]]; then
     mkdir -p ${nsroot}/etc/systemd/network
     cp -f /etc/yum/vars/* ${nsroot}/etc/yum/vars
     rpm --root=${nsroot} --import /etc/pki/rpm-gpg/*
-    yum -y --releasever=/ --installroot=${nsroot} install /usr/lib/nethserver-dc/ns-samba-*.rpm centos-release systemd-networkd bind-utils ntp | /usr/libexec/nethserver/ptrack-nsdc-install
+    yum -y --releasever=/ --installroot=${nsroot} install /usr/lib/nethserver-dc/compat-*.rpm /usr/lib/nethserver-dc/gmp-6.1.2-1.el7.x86_64.rpm /usr/lib/nethserver-dc/ns-samba-*.rpm centos-release systemd-networkd bind-utils ntp | /usr/libexec/nethserver/ptrack-nsdc-install
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-dc-install]: We could avoid hardcoding the release:
```suggestion
    yum -y --releasever=/ --installroot=${nsroot} install /usr/lib/nethserver-dc/compat-*.rpm /usr/lib/nethserver-dc/gmp-*.rpm /usr/lib/nethserver-dc/ns-samba-*.rpm centos-release systemd-networkd bind-utils ntp | /usr/libexec/nethserver/ptrack-nsdc-install
```

File: root/etc/e-smith/events/actions/nethserver-dc-upgrade [hunk 2]
```diff
@@ -33,7 +33,13 @@ rsync -ai --delete /etc/yum.repos.d/ ${nsroot}/etc/yum.repos.d
 rsync -ai --delete /etc/yum/vars/ ${nsroot}/etc/yum/vars
 
 # Upgrade chroot base system & samba package
-yum -y --releasever=/ --installroot=${nsroot} update \* /usr/lib/nethserver-dc/ns-samba-*.rpm -y | /usr/libexec/nethserver/ptrack-nsdc-install
+{
+    echo 'install' /usr/lib/nethserver-dc/compat-*.rpm
+    echo 'install' /usr/lib/nethserver-dc/gmp-6.1.2-1.el7.x86_64.rpm
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-dc-upgrade]:
```suggestion
    echo 'install' /usr/lib/nethserver-dc/gmp-*.rpm
```

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**